### ISSUE TITLE
[Feature Request] Multitask censored/custom loss

### ISSUE THREAD

**Issue by srn284** [cornellius-gp/gpytorch#1365]
With a deterministic neural network, I conduct multitask learning on some input x on 2 tasks. The basic architecture in code is:
x = featureextractor(x)
out1 = outFNN1(x)
out2 = outFNN2(x)

However, where out1 has a label, out2 does not and vice versa. So for gpytorch multitask learning I have a multitask framework copied and pasted below. But I want to select only some x for the first output and the rest for the second. Thank you

```
num_tasks=2
latentstate=17
inducingPt=16
class MultitaskGPModel(gpytorch.models.ApproximateGP):
    def __init__(self, inducing_points):
        # We have to mark the CholeskyVariationalDistribution as batch
        # so that we learn a variational distribution for each task
        variational_distribution = gpytorch.variational.CholeskyVariationalDistribution(
            inducing_points.size(-2), batch_shape=torch.Size([num_tasks]))
        # We have to wrap the VariationalStrategy in a MultitaskVariationalStrategy
        # so that the output will be a MultitaskMultivariateNormal rather than a batch output
        variational_strategy = gpytorch.variational.MultitaskVariationalStrategy(
            gpytorch.variational.VariationalStrategy(
                self, inducing_points, variational_distribution, learn_inducing_locations=True
            ), num_tasks=num_tasks)
        super().__init__(variational_strategy)
        # The mean and covariance modules should be marked as batch
        # so we learn a different set of hyperparameters
        self.mean_module = gpytorch.means.ConstantMean(batch_shape=torch.Size([num_tasks]))
        self.covar_module = gpytorch.kernels.ScaleKernel(
            gpytorch.kernels.RBFKernel(batch_shape=torch.Size([num_tasks])),
            batch_shape=torch.Size([num_tasks])
        )

    def forward(self, x):
        mean_x = self.mean_module(x)
        covar_x = self.covar_module(x)
        sout = gpytorch.distributions.MultivariateNormal(mean_x, covar_x)
        return sout
inducing_points = torch.rand(num_tasks, inducingPt, latentstate)
model = MultitaskGPModel(inducing_points)
likelihood = gpytorch.likelihoods.MultitaskGaussianLikelihood(num_tasks=num_tasks)
if torch.cuda.is_available():
    model = model.to("cuda:0")
    likelihood = likelihood.to("cuda:0")







import os
smoke_test = ('CI' in os.environ)
num_epochs = 1 if smoke_test else 20


model.train()
likelihood.train()

# We use SGD here, rather than Adam. Emperically, we find that SGD is better for variational regression
optimizer = torch.optim.Adam([
    {'params': model.parameters()},
    {'params': likelihood.parameters()},
], lr=0.01)

# Training loader
train_loader = torch.utils.data.DataLoader(torch.utils.data.TensorDataset(train_x, train_y), batch_size=10)

# Our loss object. We're using the VariationalELBO, which essentially just computes the ELBO
mll = gpytorch.mlls.VariationalELBO(likelihood, model, num_data=train_y.size(0), combine_terms=False)

# We use more CG iterations here because the preconditioner introduced in the NeurIPS paper seems to be less
# effective for VI.
for i in range(15):
    # Within each iteration, we will go over each minibatch of data
    minibatch_iter = tqdm.tqdm_notebook(train_loader, desc=f"(Epoch {i}) Minibatch")
    for x_batch, y_batch in minibatch_iter:
        optimizer.zero_grad()
#         x_batch = x_batch.reshape(17,7)
#         print(x_batch.shape)                
        x_batch, y_batch = x_batch.to("cuda:0"), y_batch.to("cuda:0")
        output = model(x_batch)
#         print(likelihood(output).mean.shape)
        log_lik, kl_div, log_prior = mll(output, y_batch)
        loss = -(log_lik - kl_div + log_prior)

        minibatch_iter.set_postfix(loss=loss.item())
        loss.backward()
        optimizer.step()

```

Of course this piece of code does multitask learning but likelihood returns loss on 2 tasks for ALL inputs. Even if i split up loss into the loglik and the kl parts, I don't know how to _not_ consider some inputs x' for loss.

Is there a way to modify likelihood internally to calculate likelihood on some x for one multitask output and the rest of the x for the other multitask output?

Also I am using:
Pytorch 1.4.0
gpytorch 1.1.1

Any way we can try and find a solution in this 1.1.1 version? 

Thank you very much and sorry if it's a repost/duplicate!
